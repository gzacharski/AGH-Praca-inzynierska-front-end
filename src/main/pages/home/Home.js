import React from "react";
import { connect } from "react-redux";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { addUser } from "../../store/model/action/creators";
import Container from "@material-ui/core/Container";
import { useStyles } from "./Home.styles";

const Home = (props) => {
  const classes = useStyles();

  const userList = (users) => users.map((user) => <h4>{user}</h4>);
  const tempUsers = ["Test user"];

  return (
    <Container maxWidth="xl" component="main" data-testid="main-container">
      <Typography variant="h5" className={classes.root} align="center">
        Strona główna
        <br />
        <Button
          variant="contained"
          color="primary"
          onClick={() => props.addUser(tempUsers[0])}
        >
          Test button
        </Button>
        {props.users && userList(props.users)}
      </Typography>
    </Container>
  );
};

const mapStateToProps = (store) => ({ users: store.modelData.users });

const mapDispatchToProps = { addUser };

export default connect(mapStateToProps, mapDispatchToProps)(Home);
