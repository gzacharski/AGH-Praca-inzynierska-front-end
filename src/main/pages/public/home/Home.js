import React from 'react';
import { PageWrapper, PublicPageTitle } from 'src/main/components/utils';

export default function Home() {
   return (
      <PageWrapper>
         <PublicPageTitle
            header="Praca inżynierska"
            subheader="System do wspomagania zarządzania placówką profilaktyki zdrowotnej"
         />
      </PageWrapper>
   );
}
