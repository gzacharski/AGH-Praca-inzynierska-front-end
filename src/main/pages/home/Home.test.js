import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Home from './Home';

const mockStore = configureStore([]);

describe('Home component', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
          modelData: {
            users: [],
          },
          stateData: {
            menuIsOpen: false,
          },
        });

        component = renderer.create(
            <Provider store={store}>
                <Home />
            </Provider>
        )
    })

    afterEach(() => {
        component.unmount()
    })

    test('should render with given state from Redux store.', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })

    test('should contain button with class name MuiButtonBase-root.', () => {
        const element=component.root.findByType('button');
        expect(element.props.className.includes('MuiButtonBase-root')).toBeTruthy();
    })

    // test('should dispatch an action on button click.', () => {
    //     renderer.act(() => {
    //         component.root.findByType('button').props.onClick();
    //     });

    //     expect(store.dispatch).toHaveBeenCalled();
    // })
})

describe('Home component rendered in DOM',()=>{
    let container;
    let store;

    beforeEach(()=>{
        container=document.createElement('div');
        store = mockStore({
            modelData: {
                users: []
            },
            stateData: {
                menuIsOpen: false
            },
        })
    })

    afterEach(()=>{
        ReactDOM.unmountComponentAtNode(container);
        container.remove();
        container=null;
    })

    test('should render',()=>{
        const element=(
            <Provider store={store}>
                <Home/>
            </Provider>
        )
        ReactDOM.render(element,container);
    })
})