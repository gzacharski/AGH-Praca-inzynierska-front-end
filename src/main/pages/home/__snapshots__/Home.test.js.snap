// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render with given state from Redux store. 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXl"
  data-testid="main-container"
>
  <h5
    className="MuiTypography-root makeStyles-root-1 MuiTypography-h5 MuiTypography-alignCenter"
  >
    Strona główna
    <br />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Test button
      </span>
    </button>
  </h5>
</main>
`;
